3rdparty/mbedtls/CMakeLists.txt

```diff
@@ -26,6 +26,11 @@ if ("${CMAKE_BUILD_TYPE}" STREQUAL "DEBUG")
   set(MBEDTLS_DEBUG_C TRUE)
 endif ()
 
+set(MBEDTLS_THREADING_C TRUE)
```

Comment:
For my understanding, can you elaborate on why these settings are being made configurable at time of CMake? 
* Why would these need to be toggled to false, particularly for the deprecation warnings? Is the threading change an optimization for single threaded optee?
* How are these used? Are they just getting defaulted to `FALSE` because they're not explicitly set in new_platforms/libmbedtls/CMakeLists.txt?
* Is there an interaction with the optee_os use of libmbedtls and its mbedtls_config_uta.h somehow?